parasolr/query.py

```diff
@@ -123,6 +127,22 @@ def count(self) -> int:
         query_opts['rows'] = 0
         return self.solr.query(**query_opts, wrap=False)['response']['numFound']
 
+    def facet_fields(self) -> dict:
+        """Return a dictionary of facets and their values/counts (also as a
+        dicitionary).
+        """
+        if self._result_cache is not None:
+            # wrap to process facets and return as dictionary
+            # for Django template support
+            qr = QueryResponse(self._result_cache)
```

Comment:
I see you're using the `QueryResponse` class to take advantage of the facet handling, that makes sense. Do you think we can adjust so that `SolrQuerySet` can use `QueryResponse` everywhere? 

I can't remember why I wasn't using it, was it because of `AttrDict`? Any thoughts on how to get around that?

Reply:
A thought, if `AttrDict` is the reason - could `QueryResponse` store both the dict and the wrapping attrdict, maybe? Then we have access to attrdict for convenience but not stuck with it.